easybuild/framework/easyblock.py

```diff
@@ -1797,6 +1798,76 @@ def sanity_check_step(self, *args, **kwargs):
         else:
             self._sanity_check_step(*args, **kwargs)
 
+    def _sanity_check_rpath(self):
+        """Sanity check binaries/libraries w.r.t. RPATH linking."""
+
+        fails = []
+
+        # hard reset $LD_LIBRARY_PATH before running RPATH sanity check
+        orig_env = env.unset_env_vars(['LD_LIBRARY_PATH'])
+
+        self.log.debug("$LD_LIBRARY_PATH during RPATH sanity check: %s", os.getenv('LD_LIBRARY_PATH', '(empty)'))
+        self.log.debug("List of loaded modules: %s", self.modules_tool.list())
+
+        not_found_regex = re.compile('not found', re.M)
+        readelf_rpath_regex = re.compile('(RPATH)', re.M)
+
+        dirpaths = [
+            os.path.join(self.installdir, 'bin'),
+            os.path.join(self.installdir, 'lib'),
+            os.path.join(self.installdir, 'lib64'),
+        ]
+        for dirpath in dirpaths:
+            if os.path.exists(dirpath):
+                self.log.debug("Sanity checking RPATH for files in %s", dirpath)
+
+                for path in [os.path.join(dirpath, x) for x in os.listdir(dirpath)]:
+                    self.log.debug("Sanity checking RPATH for %s", path)
+
+                    out, ec = run_cmd("file %s" % path, simple=False)
+                    if ec:
+                        fails.append("Failed to run 'find %s': %s" % (path, out))
+
+                    # only run ldd/readelf on dynamically linked executables/libraries
+                    # example output:
+                    # ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked (uses shared libs), ...
+                    # ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, not stripped
+                    if "dynamically linked" in out:
+                        # check whether all required libraries are found via 'ldd'
+                        out, ec = run_cmd("ldd %s" % path, simple=False)
```

Comment:
OK, but I won't block this PR for that, we can do that as a follow-up, since it'll take a while before we get the white-listed right probably.

Also, that's sort of orthogonal to `--rpath` anyway, we should always be doing that to catch unintended linking to system libraries (but then we also need to take into account `--filter-deps`, which further complicates things).

I opened an issue for this, see #1991